freqtrade/persistence/models.py

```diff
@@ -630,27 +660,24 @@ def calc_profit_ratio(self, rate: Optional[float] = None,
         return float(f"{profit_ratio:.8f}")
 
     def recalc_trade_from_orders(self):
-        # We need at least 2 entry orders for averaging amounts and rates.
-        if len(self.select_filled_orders('buy')) < 2:
-            # Just in case, still recalc open trade value
-            self.recalc_open_trade_value()
-            return
-
         total_amount = 0.0
         total_stake = 0.0
+        avg_price = None
+
         for o in self.orders:
-            if (o.ft_is_open or
-                    (o.ft_order_side != 'buy') or
-                    (o.status not in NON_OPEN_EXCHANGE_STATES)):
+            if o.ft_is_open or not o.filled:
                 continue
 
             tmp_amount = o.safe_amount_after_fee
-            tmp_price = o.average or o.price
-            if o.filled is not None:
-                tmp_amount = o.filled
+            tmp_price = o.safe_price
+            is_sell = o.ft_order_side != 'buy'
```

Comment:
Unfortunately this overlapps with the merge of `feat/short` - so there's now quite some merge conflicts - so please merge develop into your branch and fix merge-conflicts as they come up.

The problematic points will however not be the overlaps (that's where you look at) - but areas like this line (exemplary - but can happen to all other lines as well).

After the merge - please go over the "still changed lines" . Areas of focus should go to lines like this (which previously used `'buy'` or `'sell'`) - which should be updated to `trade.entry_side` and `trade.exit_side` respectively).

I don't think this will appear in too many cases (at least from a quick glance) - but it's still something to be on the lookout for to avoid nasty, difficult to track bugs (you'd increase the position instead of decreasing, potentially).

Reply:
https://github.com/freqtrade/freqtrade/blob/develop/freqtrade/persistence/models.py

`entry_side` is only used in `Order.to_json()` ; not present in Trade object

Reply:
the name is `enter_side` ... my bad.

Reply:
enter_side , entry_side (used in Order.to_json() only), entry_rate
Which is more accurate and consistence with other vars - enter or entry?

Reply:
unfortunately, both are used at the moment - and with a merge as big as the feat/short one - it was simply missed.